### PR TITLE
bzip2: 1.0.6.0.2 -> 1.0.8

### LINKED ISSUE #64817: bzip2: 1.0.7
The package has a new maintainer and upstream:

https://people.gnome.org/~federico/blog/bzip2-107-is-released.html

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Use latest upstream version, yoink updated autoconf patch from SUSE.
Might fix unpacking some very cursed files.
Dropped security patches applied upstream (see https://sourceware.org/bzip2/CHANGES).

This is probably going to explode.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
This works well enough to build libarchive (which also builds Python and curl) on x86_64-linux at least.

File: pkgs/tools/compression/bzip2/default.nix [hunk 1]
```diff
@@ -10,33 +10,24 @@
...
-    urls = map
-      (prefix: prefix + "/people/sbrabec/bzip2/tarballs/${pname}-${version}.tar.gz")
-      [
-        "http://ftp.uni-kl.de/pub/linux/suse"
-        "ftp://ftp.hs.uni-hamburg.de/pub/mirrors/suse"
-        "ftp://ftp.mplayerhq.hu/pub/linux/suse"
-        "http://ftp.suse.com/pub" # the original patched version but slow
-      ];
-    sha256 = "sha256-FnhwNy4OHe8d5M6iYCClkxzcB/EHXg0veXwv43ZlxbA=";
+    url = "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/bzip2/default.nix:16]:
```suggestion
    url = "https://sourceware.org/pub/bzip2/bzip2-${version}.tar.gz";
```

**Review by SuperSandro2000 — APPROVED** [review]
other than that LGTM

**Comment by tpwrules** [comment]
Thanks for tackling this, I have indeed run into some cursed files.

**Comment by vcunat** [comment]
Minor issues: https://github.com/NixOS/nixpkgs/pull/181708

**Pull request merged**